src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -470,6 +470,12 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync(ProjectRest
                 return false;
             }
 
+            // Log a warning if there are more than one configured source and package source mapping is not enabled
+            if (restoreRequest.Project.RestoreMetadata.Sources.Count > 1 && !restoreRequest.PackageSourceMapping.IsEnabled)
```

Comment:
I think we should raise the warning only if more than 1 http source(s) are configured. With the current logic warning is raised if there are 2 sources (1 http source + 1 local feed) configured. Please let me know if I am missing something.

Reply:
The only scenario in which you're likely to have a deterministic behavior by accident is a 2 source configuration where you have 1 local and 1 remote feed. The local feeds would get first dibs on every id being acquired, only if exact version is required. 

I don't think we should add exceptions. as Jeff mentioned you're not guaranteed determinism any time you have more than 1 feed.

Reply:
Thanks for the details. Even though source mapping is enabled, the user can configure the package patterns in such a way that one or more sources may be queried for a particular package. In this case, we may also have to raise a warning during restore saying that `multiple sources are configured in source mapping configuration for {package id} ` when CPM is enabled. May be in a follow-up PR.

Reply:
> Even though source mapping is enabled, the user can configure the package patterns in such a way that one or more sources may be queried for a particular package. In this case, we may also have to raise a warning during restore saying that multiple sources are configured in source mapping configuration for {package id} when CPM is enabled.

Personally, I'd argue this is a separate, source mapping only concern.

Reply:
In fact, it's basically what used to be strict mode, and our direction for that right now is to wait for feedback.

Reply:
I looked into it more and as long as you have a `NuGet.Config` with a `<clear />` and single `<add />` then you get a single source no matter what other configs try to inject feeds.  So I think checking for more than one feed is good.  If we get feedback later on to ignore non-HTTP feeds we can incorporate that later.  Since this a new warning, people can of course suppress it and move on.